File: Data Structures & Algorithms/subsets-ii/submission-0.py
class Solution:
    def subsetsWithDup(self, nums: List[int]) -> List[List[int]]:
        nums.sort()
        result: List[List[int]] = []
        def backtrack(start: int, current: List[int]):
            result.append(current[:])
            for index in range(start, len(nums)):
                if index > start and nums[index] == nums[index-1]:
                    continue
                current.append(nums[index])
                backtrack(index + 1, current)
                current.pop()
        backtrack(0,[])
        return result

        

File: Data Structures & Algorithms/last-stone-weight/submission-0.java
class Solution {
    public int lastStoneWeight(int[] stones) {
        int size = stones.length;

        while (size > 1) {
            Arrays.sort(stones, 0, size);

            stones[size - 2] = stones[size - 1] - stones[size - 2];
            size--;
        }

        return stones[0];
    }
}
